src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java

```diff
@@ -137,16 +145,21 @@ public Agency createAgency(final RequestToCreateAgency agencyToCreate) {
 
         final var agencyLocationType = agencyLocationTypeReferenceCodeRepository.findById(new uk.gov.justice.hmpps.prison.repository.jpa.model.ReferenceCode.Pk(AgencyLocationType.AGY_LOC_TYPE, agencyToCreate.getAgencyType())).orElseThrow(BadRequestException.withMessage(format("Agency Type [%s] not found", agencyToCreate.getAgencyType())));
 
-        final var agencyLocation = agencyLocationRepository.save(AgencyTransformer.build(agencyToCreate, agencyLocationType));
+        if (agencyLocationType.isCourt() && StringUtils.isBlank(agencyToCreate.getCourtType())) {
+            throw new EntityNotFoundException("Courts require a court type");
+        }
+        final var courtType = agencyToCreate.getCourtType() != null ? courtTypeReferenceCodeRepository.findById(new uk.gov.justice.hmpps.prison.repository.jpa.model.ReferenceCode.Pk(CourtType.JURISDICTION, agencyToCreate.getCourtType())).orElseThrow(BadRequestException.withMessage(format("Court Type [%s] not found", agencyToCreate.getCourtType()))) : null;
```

Comment:
I think this is the same as line 135? Might be worth consolidating in a private method with a nice readable name